### DISCUSSION TITLE
76. 정규화를 과도하게 할 경우 발생하는 문제를 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#78] [General]
정규화를 과도하게 할 경우 발생하는 문제를 설명해 주세요.

**Comment by linirini** [comment]
1. **과도한 조인으로 인한 성능 저하**
  필요한 데이터를 조회하려면 여러 테이블을 조인해야 하는 경우가 많아지고, 따라서 성능 저하가 발생할 수 있다.
  ****조인이 많으면 성능이 저하되는 이유**
    - 조인은 여러 테이블의 데이터를 연결하여 결과를 생성하는 작업으로 그 과정에서 각 테이블의 데이터 간 관계를 확인하고, 조건에 맞는 데이터를 비교 및 필터링하는 작업이 필요하다.
    - 조인이 많아질수록 카디널리티가 높아지므로, 데이터 비교 작업이 기하급수적으로 증가하여 성능에 부정적인 영향을 미친다.
    - 조인 과정에서 중간 결과를 생성하기 위해 메모리를 사용한다. 중간 결과가 크면 메모리 사용량이 늘어나고 디스크 I/O가 발생할 수 있다.
    - 조인에 사용되는 키에 인덱스가 없으면 데이터베이스가 조인 조건을 만족하는 행을 찾기 위해 테이블 전체를 스캔해야한다.
2. **복잡성 증가**
  테이블 관계 복잡도 증가로, 데이터베이스 구조를 이해하고 관리하기 어려워진다.
3. **쓰기 작업 성능 저하**
  데이터를 삽입하거나 수정할 때 여러 테이블에 연관 데이터를 업데이트해야 하므로 쓰기 작업 성능이 저하된다.
4. **데이터 모델링의 비효율성**
  과도한 설계를 초래할 수 있다. 예를 들어, 자주 변경되지 않는 데이터에 대해 정규화를 적용하면 불필요한 관리 작업만 늘어날 수 있다.
